arch/arm/src/rp2040/Kconfig

```diff
@@ -13,9 +20,63 @@ config RP2040_UART0
 
 if RP2040_UART0
 
+config RP2040_UART0_TX_GPIO
```

Comment:
RP2040_UART0_TX_GPIO is only used in the board file, how about we create a top level Kconfig at:
boards/arm/rp2040/Kconfig or
boards/arm/rp2040/common/Kconfig
to define this common option.

Reply:
I'm open to doing this.  

Currently a user must go the the "arch" options to turn on UART0, but then go to a completely different menu to configure the UART.  This seems wrong to me.  My goal was to keep all the pin configuration options in the same place within the menuconfig structure and since the pins are a feature of the RP2040 chip itself, it seemed better to do that in "arch" rather than "board".

If you think that all these configuration options would be better placed under the "board" menu, I'll move them all there.

Reply:
I am fine with either. @pkarashchenko @acassis what's your opinion?

Reply:
Usually peripheral driver options are out of peripheral selection menu. For example if I select `SPI0` peripheral for the "arch" I need to go to another menu to select polling vs DMA or enable other arch SPI options.
The pins are the board level configuration and not the "arch" level configuration and usually are defined in board level header file. I understand that for the boards that have many pins exposed to connectors it is reasonable to have configuration options for those pins, but I think that grouping pin configuration at board level makes more sense. At least if config of all pins are on the same screen the visual check may be done to inspect if there are not duplicates.
So my opinion is to have such config at board level

Reply:
I guess I don't understand why one would want to select SPI0 one place and have to go somewhere else to configure it.  That's one thing that I find particularly annoying in the NuttX configuration tool.  

I could understand having these configurations at the board level if the configuration tool had a convenient way to limit numeric input to only a specific number of discreet choices, so that unimplemented choices could not be made. But even then, I would think both selection and configuration should be in the same place.

Is there some advantage to the end user for having the configuration options split up?

Anyway, like I said, I'm open to moving the configuration to the board menu.

Reply:
It is a layered approach.

From a schematic we get "what" is wired

from a SW use case we get "How" it is used

Reply:
I moved the pin definitions to boards/arm/rp2040/common/Kconfig as requested.